### DISCUSSION TITLE
[Chapter 08] spring-boot-starter-data-jpa 의존성에는 AOP 가 포함되어 있다?

### DISCUSSION THREAD

**Discussion by corock** [cowaine/spring-boot-for-msa#89] [Chapter 08. Data Persistence] [answered]
왜 spring-boot-starter-data-jpa 의존성에 AOP 가 있을까?

> spring-boot-starter-data-jpa 에는 다음과 같은 주요 라이브러리가 포함되어 있다.
> - spring-boot-starter-aop: 스프링 AOP 를 사용하는 스프링 부트 스타터
> - (...)
>
> (8.3 Spring Data JPA 기능과 설정, p.  463)

**Comment by BvrPark** [comment]
이게 답변이 될지는 모르겠지만 `@Transactional` 어노테이션이 스프링의 AOP기능을 활용하여 트랜잭션 관리를 단순화한다라고 알고있는데
JPA에서 이런식으로 AOP기능을 활용한 어노테이션을 사용해야 될 수 밖에 없어서 의존성에 AOP가 있는게 아닐까요?

**Comment by coalong** [comment, marked as answer]
spring-boot-starter-data-jpa 안에 포함되어있는 라이브러리를 살펴보면거 진짜 섬세한 것 같습니다!

AOP 를 사용하는 이유를 책을 읽기만 했을 때 유추한 것은

Audit 감사 기능이나 Spring Data JPA 를 이벤트로 받아서 처리하는 부분(아직 안나옴) 에서 사용할 것 같다?는 생각을 했습니다.

**Comment by Irisation23** [comment]
### 질의 응답 이외의 CheckPoint

1. 질문의도는 jpa에 AOP 적용된 부분이 있을까?

**Comment by JoisFe** [comment]
![Uploading image.png…]()
